libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -107,12 +103,13 @@ internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
     internal const val LOCATION_BUFFER_MAX_SIZE = 20
     private const val ONE_SECOND = 1000
     private const val MOCK_PROVIDER =
-        "com.mapbox.navigation.core.replay.route.ReplayRouteLocationEngine"
+        "com.mapbox.navigation.core.replay.ReplayLocationEngine"
 
     internal const val TAG = "MAPBOX_TELEMETRY"
     private const val EVENT_VERSION = 7
     private lateinit var context: Context // Must be context.getApplicationContext
     private lateinit var telemetryThreadControl: JobControl
+    private val telemetryScope: CoroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.IO)
```

Comment:
Hmm interesting https://proandroiddev.com/kotlin-coroutines-patterns-anti-patterns-f9d12984c68e

Reply:
`TIL` Looks like this is how we're handling the crash. Because SupervisorJob watches over multiple children launched within scope, and prevents one child from crashing another.

Does this mean we've moved error handling to somewhere else?

Reply:
What's the difference between telemetryThreadControl and telemetryScope?

Will we be migrating to telemetryScope?

Reply:
This was mainly added to support running operations that shouldn’t be cancelled in Coroutines even if its parent is cancelled https://medium.com/androiddevelopers/coroutines-patterns-for-work-that-shouldnt-be-cancelled-e26c40f142ad That was the issue and how we're trying to solve it here as described in https://github.com/mapbox/mapbox-navigation-android/pull/3162#issue-433390967 In other words, we create a separate scope that we know for sure it's not going to be canceled so work is guaranteed to run with `join()`. Does that make sense?

Reply:
> What's the difference between telemetryThreadControl and telemetryScope?

`telemetryThreadControl` 👉 https://github.com/mapbox/mapbox-navigation-android/blob/68758c5dcfebadf0b296304fdbcae4749114fd83/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L203

`telemetryScope` 👉 separate scope that we know for sure it's not going to be canceled

> Will we be migrating to telemetryScope?

Nah, that's on purpose and needed as described above. We need to revisit `ThreadController` implementation though https://github.com/mapbox/mapbox-navigation-android/issues/3156

Reply:
Yep I get it 🤔  

telemetryThreadControl is destroyed by onDestroy
telemetryScope will keep going

this class feels like we'll solve one problem and create a new one. populateNavigationEvent gets executed way late and sends bogus data? 

👍  fix crash